app/controllers/AccountManagement.scala

```diff
@@ -71,19 +73,20 @@ object AccountManagement extends Controller with LazyLogging {
     val errorCodes = errorCode.toSeq.flatMap(_.split(',').map(_.trim)).filterNot(_.isEmpty).toSet
 
     (for {
-      sub <- OptionT(subscription)
+      subEither <- OptionT(subscription)
+      sub = subEither.fold(identity, identity)
       allHolidays <- OptionT(tpBackend.suspensionService.getHolidays(sub.name).map(_.toOption))
-      billingSchedule <- OptionT(tpBackend.commonPaymentService.billingSchedule(sub, 12))
-      suspendableDays = Config.suspendableWeeks * sub.plan.products.without(Delivery).size
-    } yield {
-      sub.plan.products.seq.contains(Delivery).fold({
+      billingSchedule <- OptionT(tpBackend.commonPaymentService.billingSchedule(sub.id, numberOfBills = 12))
+      suspendableDays = Config.suspendableWeeks * sub.plan.charges.benefits.size
+    } yield
+      subEither.fold({ s =>
```

Comment:
seems github doesn't hide comments on outdated diffs now, but I fixed this in 4cf5501